Source/WebCore/html/HTMLSummaryElement.cpp

```diff
@@ -96,12 +98,15 @@ bool HTMLSummaryElement::isActiveSummary() const
     return details->isActiveSummary(*this);
 }
 
-static bool isClickableControl(EventTarget* target)
+static bool isInSummaryInteractiveContent(EventTarget* target)
 {
     if (!is<Element>(target))
         return false;
-    auto& element = downcast<Element>(*target);
-    return is<HTMLFormControlElement>(element) || is<HTMLFormControlElement>(element.shadowHost());
+    for (auto* node = &downcast<Node>(*target); node && !is<HTMLSummaryElement>(node); node = node->parentOrShadowHostNode()) {
```

Comment:
I realize we have `parentOrShadowHostElement` which is probably more appropriate:

```suggestion
    for (auto* element = downcast<Element>(target); element && !is<HTMLSummaryElement>(element); element = element->parentOrShadowHostElement()) {
```